mpi-proxy-split/seq_num.cpp

```diff
@@ -170,16 +182,15 @@ void commit_begin(MPI_Comm comm) {
     // Stop before the critical section during checkpoint time.
     if (ckpt_pending && check_seq_nums()) {
       current_phase = STOP_BEFORE_CS;
-      while (!freepass && ckpt_pending);
-      freepass = false;
+      while (ckpt_pending && sem_trywait(&freepass_sem) != 0);
```

Comment:
Can we replace this by:
```
if (ckpt_pending) {
  sem_wait(&freepass_sem);
}
```
It's more like how we say it in English.  And when I see `sem_trywait()`, I assume the standard sem_post/sem_wait isn't sufficient, and we need an extension to the normal concept.  But here, the standard semaphore operations are enough.